### DISCUSSION TITLE
DataObject with custom DataFormat is always "Context"

### DISCUSSION THREAD

**Discussion by GundlackFlorian** [AvaloniaUI/Avalonia#12917] [General]
Hello,
I have a little question, i'm using Drag&Drop, and when i add a object with a custom DataFormat, when dropped the DataFormat is always "Context" not the name i set
For example:
```
[...]
var dragData = new DataObject();
dragData.Set( "TestFormat", itemViewModel );
Task.Run( async () => await DragDrop.DoDragDrop( e, dragData, DragDropEffects.Copy | DragDropEffects.Move | DragDropEffects.Link ) );
```

Then when i print the data format list received in the DropEvent:
```
AddHandler( DragDrop.DropEvent, ( sender, args ) =>
{
    if ( args.Data is DataObject data )
    {
        Trace.WriteLine( "DataFormats:" );
        foreach ( string dataFormat in data.GetDataFormats() )
        {
            Trace.WriteLine( $"\tFormat: \"{dataFormat}\"" );
        }
    }
}
```

It prints
``` 
DataFormats:
	Format: "Context"
```
	
The data is valid, but is this normal behavior ?
Thanks

Edit: 

Also if i print the DataFormat before sending them with DoDragDrop their have valid name.

OS: Windows
Avalonia 11.0.4

**Comment by kekekeks** [comment]
Which OS?

**Reply by GundlackFlorian** [reply]
Sorry i should have precised: Windows

**Comment by GundlackFlorian** [comment]
Oh i just found out why, it's related to Avalonia Behaviors, i didn't noticed that it uses it's own DoDragDrop with DataFormat set as "Context", i guess it was overriding my one.
Closing the issue.
